Documentation/dev_workflow.md

```diff
@@ -0,0 +1,156 @@
+This document shows an example of development process for Xen and GRUB installed
+on remote machine.
+
+Assumptions:
+
+- during development, you may and probably at some point will break booting on
+  target device, so don't test on host
+- you should have a way of booting another Linux distribution on target, even if
+  main disk's bootloader isn't working
+  - this can be either another drive with another bootloader or iPXE
+  - bootable USB stick with live Debian should be good
+- there is network connection between target and host
+  - target obtains its IP through DHCP or is set manually (not shown below)
+
+# Workflow
+
+## Cloning and starting Docker container
+
+If you're testing or working on specific changes, use appropriate branches
+instead of main development ones.
+
+```shell
+git clone git@github.com:TrenchBoot/grub.git
+cd grub
+git checkout intel-txt-mb2
+cd ..
+git clone git@github.com:3mdeb/xen.git
+cd xen
+git checkout aem/develop
+cd ..
+docker run --rm -it -v $PWD/grub:/home/trenchboot/grub \
+-v $PWD/xen:/home/trenchboot/xen -w /home/trenchboot \
+ghcr.io/trenchboot/trenchboot-sdk:master /bin/bash
+```
+
+## Building
+
+These steps are done in Docker. There are some redundant `cd`s which can be
+optimized, they are listed to make each step more or less standalone.
+
+### GRUB - one time configuration
+
+```shell
+cd grub
+./bootstrap
+./autogen.sh
+mkdir -p build && cd build
+../configure --disable-werror --prefix=$PWD/local
+cd ../..
+```
+
+### GRUB - (re)build
+
+```shell
+cd grub/build
+make
+make install
+cd ../..
+```
+
+### Xen
+
+```shell
+cd xen
+make build-xen
+cd ..
+```
+
+## Installation
+
+### Host side, outside of Docker
+
+This step prepares a list of files that must be copied to target and starts a
+simple HTTP server. Substitute `<<hostIP>>` with your address.
+
+```shell
+ls -1 grub/build/local/lib/grub/i386-pc/ | \
```

Comment:
```suggestion
export HOST_IP="<<hostIP>>"
ls -1 grub/build/local/lib/grub/i386-pc/ | \
```

Reply:
I think this way it would be easier to just copy-paste following commands, rather than copying and chaging IP somewhere inside them.

Reply:
Changed.